webdriver/tests/bidi/script/call_function/arguments.py

```diff
@@ -0,0 +1,76 @@
+import pytest
+
+from webdriver.bidi.modules.script import ContextTarget
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "argument",
+    [
+        {"type": "undefined"},
+        {"type": "null"},
+        {"type": "string", "value": "foobar"},
+        {"type": "string", "value": "2"},
+        {"type": "number", "value": "NaN"},
+        {"type": "number", "value": "-0"},
+        {"type": "number", "value": "Infinity"},
+        {"type": "number", "value": "-Infinity"},
+        {"type": "number", "value": 3},
+        {"type": "number", "value": 1.4},
+        {"type": "boolean", "value": True},
+        {"type": "boolean", "value": False},
+        {"type": "bigint", "value": "42"},
+    ],
+)
+async def test_primitive_values(bidi_session, top_context, argument):
+    result = await bidi_session.script.call_function(
+        function_declaration=f"(arg) => arg",
```

Comment:
So the problem here is that we actually do not compare the result of the deserialization but a complete round trip including deserialization and serialization. In case of the serialization also be broken we wouldn't detect a failure for the deserialization.

As such we would need an assertion for a type check as part of the function declaration. If that passes the value also needs to be checked. Then the test will pass if no `exceptionDetails` are part of the return value for the `script.callFunction` call.

The same actually also applies to the other test.

Reply:
@sadym-chromium fyi whimboo is off for a few weeks, but I agree with his feedback here. Ideally this should only assert deserialization. I received a ping for this review today but I didn't see a new patch. I'm assuming the ping was sent before seeing the comment above.

Reply:
Yeah, I pinged before I saw the comment.

Ideally, we want to test only deserialization, but comparing sets, arrays, maps and objects on the backend is quite messy. So having we tested serialization, I guess it's acceptable trade-off just to check the types of non-pimitives on the server side, and check the "deserialized-serialized" values on the client side.

Reply:
@juliandescottes